src/libdredd/src/mutation_replace_unary_operator.cc

```diff
@@ -221,12 +230,36 @@ std::string MutationReplaceUnaryOperator::GenerateMutatorFunction(
 
   return new_function.str();
 }
+
+bool MutationReplaceUnaryOperator::IsRedundantReplacementOperator(
+    clang::UnaryOperatorKind op, clang::ASTContext& ast_context) const {
+  if (MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), 0,
+                                              ast_context) ||
+      MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), 1,
+                                              ast_context)) {
+    if (op == clang::UO_Minus || op == clang::UO_LNot) {
+      return true;
+    }
+  }
+
+  if (MutationReplaceExpr::ExprIsEquivalentTo(*unary_operator_.getSubExpr(), -1,
+                                              ast_context) &&
+      op == clang::UO_Minus) {
+    return true;
+  }
+
+  return false;
+}
```

Comment:
Can you add some comments to justify these cases? I can see tha -0 is equivalent, and -1 is covered by replacing with -1 in the signed case (do we apply - to unsigned?)

In general some comments would be good.

Reply:
Added comments.

Reply:
@afd I was concerned about that being the case, I think i noticed in one of the test cases that in an expression like `unsigned int x = ~5`, the expression may be replaced with a negative number. I'll take a look and check.

Reply:
I have filled this as an issue #115.